website/docs/language/resources/ephemeral/write-only.mdx

```diff
@@ -1,44 +1,128 @@
 ---
 page_title: Use write-only arguments
-description: Learn how to use write-only arguments to set temporary values that can only be overwritten and are not stored in Terraform's state.
+description: Learn how to use write-only arguments to set temporary values that are not stored in Terraform's state or plan files.
 ---
 
-<!-- THIS IS HIDDEN FOR NOW - TODO FOR GA: unhide when we have a code snippet to share and I can add more details -->
-
 # Use write-only arguments
 
-Terraform resources can include ephemeral arguments, also known as attributes, for data that only needs to exist temporarily. An ephemeral argument on a resource is called a "write-only argument". Write-only arguments can help store generated sensitive data for the current Terraform operation, such as a short-lived password, token, or session identifier.
+Write-only arguments let you securely pass temporary values to Terraform resources during a Terraform operation without persisting those values to state or plan files. Use write-only arguments to store sensitive data for the current Terraform operation, such as a short-lived password, token, or session identifier. 
+
+## Background
+
+Write-only arguments are only available during the runtime of a Terraform operation, and Terraform omits those values from state and plan files.
+
+> **Hands-on**: Declare a write-only argument in the [Upgrade RDS major version](/terraform/tutorials/aws/rds-upgrade) tutorial.
+
+You can use write-only arguments to pass ephemeral values directly to a resource without Terraform persisting that value. For example, you can directly pass a generated password or a fetched token to a write-only argument without risking Terraform exposing that value. Use write-only arguments to ensure sensitive data only exists during the current Terraform operation.
```

Comment:
This might be re-iterating what the intro says a bit too much? Maybe the intro should be more focused on the "why" and this para more on the "how"?

Reply:
Lol thank you - you can tell I rewrote both of these way too many times 😂

I kept the intro the same because I do feel like it covers the "why" pretty well, but expanded the background to explain a bit more of how you can use ephemeral values _together_. Which I think is the context that I was missing here, and what fits best in a background section